test/unit/Site.test.js

```diff
@@ -189,6 +189,32 @@ test('Site read site config for custom site config', async () => {
   expect(site.siteConfig).toEqual(customSiteJson);
 });
 
+test('Site resolves variables referencing other variables', async () => {
+  const json = {
+    'lib/asset/glyphicons.csv': '',
+    'lib/template/page.ejs': PAGE_EJS,
+    'site.json': SITE_JSON_DEFAULT,
+    '_markbind/variables.md':
+    '<span id="level1">variable</span>'
+    + '<span id="level2">{{level1}}</span>'
+    + '<span id="level3">{{level2}}</span>'
+    + '<span id="level4">{{level3}}</span>',
+  };
+  fs.vol.fromJSON(json, '');
```

Comment:
Could we account for different variables and glyphicons (if possible)? Also could we add sub sites as well?

Reply:
> Could we account for different variables and glyphicons (if possible)?

What do you mean by different variables? Do you mean level4 referring to both level3 and level2?

Will modify the test to include glyphicons.

> Also could we add sub sites as well?

The current implementation of MarkBind does not allow cross-referencing variables from other sub sites, does it? Anyway that seems to be out of scope of the current PR.

Reply:
> Could we account for different variables and glyphicons (if possible)?

Oh i meant to say adding another word instead of 'variable'. Perhaps changing level3 to a glyphicon?

> The current implementation of MarkBind does not allow cross-referencing variables from other sub sites, does it? Anyway that seems to be out of scope of the current PR.

It doesnt. I was just concerned if there was any unwanted new behaviour was introduced where a subsite variable can reference an outersite. As for adding this new behaviour, i agree this is out of scope.